app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CreateEventActivity.kt

```diff
@@ -242,17 +291,67 @@ class CreateEventActivity : ActivityForAccessingTumOnline<CreateEventResponse>(R
         }
         event.description = description
         this.event = event
+        events = generateEvents()
+
+        for (curEvent in events!!) {
+            val apiCall = apiClient.createEvent(curEvent, null)
+            fetch(apiCall)
+        }
+    }
 
-        val apiCall = apiClient.createEvent(event, null)
-        fetch(apiCall)
+    /**
+     * generates a list of events that are added to the calendar
+     * depending on the repeat-setting
+     */
+    private fun generateEvents(): List<CalendarItem> {
+        if (!repeats) {
+            return listOf(this.event!!)
+        }
+        val items = ArrayList<CalendarItem>()
+        seriesId = generateSeriesId()
+
+        // event ends after n times
+        if (endAfterRadioBtn.isChecked) {
+            val numberOfEvents = eventRepeatsTimes.text.toString().toInt()
+            for (i in 0 until numberOfEvents) {
+                val item = CalendarItem()
+                item.title = event!!.title
+                item.description = event!!.description
+                item.dtstart = event!!.dtstart.plusWeeks(i)
+                item.dtend = event!!.dtend.plusWeeks(i)
+                items.add(item)
+            }
+            // event ends after "last" date
+        } else {
+            var curDateStart = event!!.dtstart
+            var curDateEnd = event!!.dtend
+            last.plusDays(1)
+            while (curDateStart.isBefore(last)) {
+                val item = CalendarItem()
+                item.title = event!!.title
+                item.description = event!!.description
+                item.dtstart = curDateStart
+                item.dtend = curDateEnd
+                curDateStart = curDateStart.plusWeeks(1)
+                curDateEnd = curDateEnd.plusWeeks(1)
+                items.add(item)
+            }
+        }
+        return items
     }
 
     override fun onDownloadSuccessful(response: CreateEventResponse) {
-        event?.let {
+        events?.get(apiCallsFetched++)?.let {
```

Comment:
So if you are sure that the fetch calls are synchronous, this works. Are you sure? ^^